pkg/common/agentpathtemplate/template.go

```diff
@@ -3,14 +3,164 @@ package agentpathtemplate
 import (
 	"bytes"
 	"text/template"
+
+	sprig "github.com/Masterminds/sprig/v3"
 )
 
+var funcList = [...]string{
+	"abbrev",
+	"abbrevboth",
+	"trunc",
+	"trim",
+	"upper",
+	"lower",
+	"title",
+	"untitle",
+	"substr",
+	"repeat",
+	"trimAll",
+	"trimSuffix",
+	"trimPrefix",
+	"nospace",
+	"initials",
+	"randAlphaNum",
+	"randAlpha",
+	"randAscii",
+	"randNumeric",
+	"swapcase",
+	"shuffle",
+	"snakecase",
+	"camelcase",
+	"kebabcase",
+	"wrap",
+	"wrapWith",
+	"contains",
+	"hasPrefix",
+	"hasSuffix",
+	"quote",
+	"squote",
+	"cat",
+	"indent",
+	"nindent",
+	"replace",
+	"plural",
+	"sha1sum",
+	"sha256sum",
+	"sha512sum",
+	"adler32sum",
+	"toString",
+	"seq",
+	"splitList",
+	"toStrings",
+	"join",
+	"sortAlpha",
+	"default",
+	"empty",
+	"coalesce",
+	"all",
+	"any",
+	"compact",
+	"mustCompact",
+	"fromJson",
+	"toJson",
+	"toPrettyJson",
+	"toRawJson",
+	"mustFromJson",
+	"mustToJson",
+	"mustToPrettyJson",
+	"mustToRawJson",
+	"ternary",
+	"deepCopy",
+	"mustDeepCopy",
+	"typeOf",
+	"typeIs",
+	"typeIsLike",
+	"kindOf",
+	"kindIs",
+	"deepEqual",
+	"base",
+	"dir",
+	"clean",
+	"ext",
+	"isAbs",
+	"b64enc",
+	"b64dec",
+	"b32enc",
+	"b32dec",
+	"tuple",
+	"list",
+	"dict",
+	"get",
+	"set",
+	"unset",
+	"hasKey",
+	"pluck",
+	"keys",
+	"pick",
+	"omit",
+	"merge",
+	"mergeOverwrite",
+	"mustMerge",
+	"mustMergeOverwrite",
+	"values",
+	"append",
+	"push",
+	"mustAppend",
+	"mustPush",
+	"prepend",
+	"mustPrepend",
+	"first",
+	"mustFirst",
+	"rest",
+	"mustRest",
+	"last",
+	"mustLast",
+	"initial",
+	"mustInitial",
+	"reverse",
+	"mustReverse",
+	"uniq",
+	"mustUniq",
+	"without",
+	"mustWithout",
+	"has",
+	"mustHas",
+	"slice",
+	"mustSlice",
+	"concat",
+	"dig",
+	"chunk",
+	"mustChunk",
+	"uuidv4",
+	"fail",
+	"regexMatch",
+	"mustRegexMatch",
+	"regexFindAll",
+	"mustRegexFindAll",
+	"regexFind",
+	"mustRegexFind",
+	"regexReplaceAll",
+	"mustRegexReplaceAll",
+	"regexReplaceAllLiteral",
+	"mustRegexReplaceAllLiteral",
+	"regexSplit",
+	"mustRegexSplit",
+	"regexQuoteMeta",
+	"urlParse",
+	"urlJoin",
+}
+
 // Parse parses an agent path template. It changes the behavior for missing
 // keys to return an error instead of the default behavior, which renders a
 // value that requires percent-encoding to include in a URI, which is against
 // the SPIFFE specification.
 func Parse(text string) (*Template, error) {
-	tmpl, err := template.New("agent-path").Option("missingkey=error").Parse(text)
+	sprigMap := sprig.TxtFuncMap()
+	ourMap := make(template.FuncMap)
+	for _, f := range funcList {
+		ourMap[f] = sprigMap[f]
```

Comment:
We should fail if the expected function does not exist in sprigMap (e.g. we had a typo or a function was removed)